src/act/mod.rs

```diff
@@ -0,0 +1,274 @@
+use func::Func;
+use proc_macro2::TokenStream;
+use quote::quote;
+use std::collections::HashMap;
+use std::hash::Hash;
+
+use self::{
+    node::{
+        canister_method::{
+            CanisterMethod,
+            {
+                HeartbeatMethod, InitMethod, InspectMessageMethod, PostUpgradeMethod,
+                PreUpgradeMethod, QueryMethod, UpdateMethod,
+            },
+        },
+        data_type::{func, Record, Tuple, TypeAlias, Variant},
+        DataType, Node, NodeContext, {ExternalCanister, FunctionGuard},
+    },
+    proclamation::{Proclaim, Proclamation},
+};
+use crate::generators::{candid_file_generation, random, vm_value_conversion};
+
+pub mod node;
+pub mod proclamation;
+pub mod to_node;
+
+/// An easily traversable representation of a rust canister
+pub struct AbstractCanisterTree {
+    pub cdk_name: String,
+    pub canister_methods: CanisterMethods,
+    pub data_types: DataTypes,
+    pub external_canisters: Vec<ExternalCanister>, // TODO Make sure these are collected as children
+    pub function_guards: Vec<FunctionGuard>,
```

Comment:
Should we move this into the list of canister methods?

Reply:
Also rename to be guard_function

Reply:
> Should we move this into the list of canister methods?

No, it's not a canister method. We decided that the benefit of fitting it in there wasn't worth it. 
It became clear as we were going down the list of canister methods, 
init
heartbeat
preupgrade
postupgrade
query
update
*hesitation*
guard function 

it didn't feel right so we nixed it.

The renaming has taken place though